File: src/main/java/com/now/nowbot/throwable/TipsException.java
package com.now.nowbot.throwable;

import java.util.Objects;

public class TipsException extends Exception implements BotException {
    String message;
    byte[] image = null;

    public TipsException(String message) {
        setMessage(message);
    }

    public TipsException(String message, Object... args) {
        setMessage(String.format(message, args));
    }

    public TipsException(byte[] image) {
        this.image = image;
    }

    public boolean hasImage() {
        return Objects.nonNull(image);
    }

    public byte[] getImage() {
        return image;
    }

    public void setImage(byte[] image) {
        this.image = image;
    }

    @Override
    public String getMessage() {
        if (message != null) {
            return message;
        }
        return super.getMessage();
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: src/main/java/com/now/nowbot/throwable/TipsException.kt
package com.now.nowbot.throwable

open class TipsException : Exception, BotException {
    override var message: String? = null
        get() {
            return if (field != null) {
                field!!
            } else {
                super.message!!
            }
        }
    final override var image: ByteArray? = null

    constructor()

    constructor(message: String?) {
        this.message = message
    }

    constructor(message: String, vararg args: Any?) {
        this.message = String.format(message, *args)
    }

    constructor(image: ByteArray?) {
        this.image = image
    }

    override fun hasImage(): Boolean {
        return image != null
    }
}
